### PR TITLE
Fix RejectException

### PR DESCRIPTION
**Pull request by finefuture**
## What is the purpose of the change
image
 If executor was shutdown, shouldn't submit task.

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Review by AlbumenJ — COMMENTED** [review]
@guohao @EarthChen @icodening PTAL

File: dubbo-common/src/main/java/org/apache/dubbo/common/threadpool/serial/SerializingExecutor.java [hunk 1]
```diff
@@ -68,8 +69,10 @@ private void schedule(Runnable removable) {
         if (atomicBoolean.compareAndSet(false, true)) {
             boolean success = false;
             try {
-                executor.execute(this);
-                success = true;
+                if (!executor.isShutdown()) {
```
Comment (icodening) [inline dubbo-common/src/main/java/org/apache/dubbo/common/threadpool/serial/SerializingExecutor.java:72]: maybe only needed `instanceof` and `cast`, other code should not change

**Review by icodening — CHANGES_REQUESTED** [review]

**Review by EarthChen — APPROVED** [review]
LGTM

**Review by AlbumenJ — APPROVED** [review]

**Comment by bert82503** [comment]
we use `dubbo-3.2.5`, encounter the same issue.

```log
message:[DUBBO] Meet Exception on ClientResponseHandler, status code is: INTERNAL, dubbo version: 3.2.5, current host: *, error code: 4-10. This may be caused by , go to https://dubbo.apache.org/faq/4/10 to find instructions.
logger:org.apache.dubbo.rpc.protocol.tri.transport.TripleHttp2ClientResponseHandler
level:WARN
trace_id:N/A
exception:java.util.concurrent.RejectedExecutionException: null at org.apache.dubbo.common.threadpool.ThreadlessExecutor.execute(ThreadlessExecutor.java:101) at org.apache.dubbo.common.threadpool.serial.SerializingExecutor.schedule(SerializingExecutor.java:71) at org.apache.dubbo.common.threadpool.serial.SerializingExecutor.execute(SerializingExecutor.java:64) at org.apache.dubbo.rpc.protocol.tri.stream.TripleClientStream$ClientTransportListener.onHeader(TripleClientStream.java:433) at org.apache.dubbo.rpc.protocol.tri.transport.TripleHttp2ClientResponseHandler.channelRead0(TripleHttp2ClientResponseHandler.java:65) at org.apache.dubbo.rpc.protocol.tri.transport.TripleHttp2ClientResponseHandler.channelRead0(TripleHttp2ClientResponseHandler.java:35) at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:99) at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440) at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) at io.netty.handler.codec.http2.AbstractHttp2StreamChannel$Http2ChannelUnsafe.doRead0(AbstractHttp2StreamChannel.java:939) at io.netty.handler.codec.http2.AbstractHttp2StreamChannel.fireChildRead(AbstractHttp2StreamChannel.java:584) at io.netty.handler.codec.http2.Http2MultiplexHandler.channelRead(Http2MultiplexHandler.java:186) at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442) at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) at io.netty.handler.codec.http2.Http2FrameCodec.onHttp2Frame(Http2FrameCodec.java:712) at io.netty.handler.codec.http2.Http2FrameCodec$FrameListener.onHeadersRead(Http2FrameCodec.java:644) at io.netty.handler.codec.http2.Http2FrameCodec$FrameListener.onHeadersRead(Http2FrameCodec.java:638) at io.netty.handler.codec.http2.Http2FrameListenerDecorator.onHeadersRead(Http2FrameListenerDecorator.java:48) at io.netty.handler.codec.http2.Http2EmptyDataFrameListener.onHeadersRead(Http2EmptyDataFrameListener.java:63) at io.netty.handler.codec.http2.DefaultHttp2ConnectionDecoder$FrameReadListener.onHeadersRead(DefaultHttp2ConnectionDecoder.java:409) at io.netty.handler.codec.http2.DefaultHttp2ConnectionDecoder$FrameReadListener.onHeadersRead(DefaultHttp2ConnectionDecoder.java:337) at io.netty.handler.codec.http2.Http2InboundFrameLogger$1.onHeadersRead(Http2InboundFrameLogger.java:56) at io.netty.handler.codec.http2.DefaultHttp2FrameReader$2.processFragment(DefaultHttp2FrameReader.java:476) at io.netty.handler.codec.http2.DefaultHttp2FrameReader.readHeadersFrame(DefaultHttp2FrameReader.java:484) at io.netty.handler.codec.http2.DefaultHttp2FrameReader.processPayloadState(DefaultHttp2FrameReader.java:253) at io.netty.handler.codec.http2.DefaultHttp2FrameReader.readFrame(DefaultHttp2FrameReader.java:159) at io.netty.handler.codec.http2.Http2InboundFrameLogger.readFrame(Http2InboundFrameLogger.java:41) at io.netty.handler.codec.http2.DefaultHttp2ConnectionDecoder.decodeFrame(DefaultHttp2ConnectionDecoder.java:173) at io.netty.handler.codec.http2.DecoratingHttp2ConnectionDecoder.decodeFrame(DecoratingHttp2ConnectionDecoder.java:63) at io.netty.handler.codec.http2.Http2ConnectionHandler$FrameDecoder.decode(Http2ConnectionHandler.java:393) at io.netty.handler.codec.http2.Http2ConnectionHandler.decode(Http2ConnectionHandler.java:453) at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:529) at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:468) at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:290) at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440) at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166) at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788) at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724) at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650) at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562) at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) at java.lang.Thread.run(Thread.java:750)
```

File: dubbo-common/src/main/java/org/apache/dubbo/common/threadpool/serial/SerializingExecutor.java [hunk 2]
```diff
@@ -42,7 +42,7 @@ public final class SerializingExecutor implements Executor, Runnable {
      */
     private final AtomicBoolean atomicBoolean = new AtomicBoolean();
 
-    private final ExecutorService executor;
```
Comment (bert82503) [inline dubbo-common/src/main/java/org/apache/dubbo/common/threadpool/serial/SerializingExecutor.java]: Why change type from ExecutorService to Executor?
I think there's no need to change it.

File: dubbo-common/src/main/java/org/apache/dubbo/common/threadpool/serial/SerializingExecutor.java [hunk 3]
```diff
@@ -41,7 +42,7 @@ public final class SerializingExecutor implements Executor, Runnable {
      */
     private final AtomicBoolean atomicBoolean = new AtomicBoolean();
 
-    private final Executor executor;
```
Comment (bert82503) [inline dubbo-common/src/main/java/org/apache/dubbo/common/threadpool/serial/SerializingExecutor.java]: Is there rollback type?
nice work.

Comment (finefuture) [reply dubbo-common/src/main/java/org/apache/dubbo/common/threadpool/serial/SerializingExecutor.java]: Yep, already rolled back.

File: dubbo-common/src/main/java/org/apache/dubbo/common/threadpool/serial/SerializingExecutor.java (hunk 2 above)
Comment (finefuture) [reply dubbo-common/src/main/java/org/apache/dubbo/common/threadpool/serial/SerializingExecutor.java]: Already rolled back.

**Review by icodening — APPROVED** [review]
LGTM

**Pull request merged**